src/sinks/aws_kinesis/firehose/record.rs

```diff
@@ -46,14 +48,24 @@ impl SendRecord for KinesisFirehoseClient {
     type T = KinesisRecord;
     type E = KinesisError;
 
-    async fn send(&self, records: Vec<Self::T>, stream_name: String) -> Option<SdkError<Self::E>> {
+    async fn send(
+        &self,
+        records: Vec<Self::T>,
+        stream_name: String,
+    ) -> Result<KinesisResponse, SdkError<Self::E>> {
+        let rec_count = records.len().clone();
+
         self.client
             .put_record_batch()
             .set_records(Some(records))
             .delivery_stream_name(stream_name)
             .send()
             .instrument(info_span!("request").or_current())
             .await
-            .err()
+            .map(|output: PutRecordBatchOutput| KinesisResponse {
+                count: rec_count,
+                failure_count: output.failed_put_count().unwrap_or(0) as usize,
+                events_byte_size: 0,
```

Comment:
the events size isn't available here. Wasn't sure the best way to modify this - will think about it. I may just return the failure count for now, and build the response in the `Service`